Bizyhood/Meta.php

```diff
@@ -0,0 +1,168 @@
+
+        if ($business->claimed == 1 && $business->description != '') {
+          $metadata['description'] = wp_trim_words(htmlentities($business->description), Bizyhood_Core::META_DESCRIPTION_LENGTH, '');
+        } else {
+          $metadata['description'] = htmlentities($business->name.' is a hyper-local, small business, located in and/or serving the '. $business->locality .', '. $business->region .' area.');
+        }
+        
+        
+        if($business->business_logo) {
+          $metadata['logo'] = $business->business_logo->image->url;
+        }
+      }
+      
+      // overview page metadata
+      if (is_page($events_page) && isset($wp_query->query_vars['bizyhood_id']) && $wp_query->query_vars['bizyhood_id'] != '') {
+
+        $bizyhood_name = '';
+        
+        $bizyhood_name = urldecode($wp_query->query_vars['bizyhood_name']);
+        $bizyhood_id = urldecode($wp_query->query_vars['bizyhood_id']);
+        
+        $single_event_information = Bizyhood_ApiCalls::single_business_additional_info('events', $bizyhood_name, $bizyhood_id);
+        
+        $business = '';
+        
+        if($single_event_information === false || !isset($single_event_information->name)) {
+          return $buffer;
+        }
+        
+        $metadata['title'] = htmlentities($single_event_information->name .', '. $single_event_information->business_name .' - '.get_bloginfo('name'));
+        $metadata['canonical'] = get_permalink($events_page) . $wp_query->query_vars['bizyhood_name'] .'/'.$wp_query->query_vars['bizyhood_id'].'/';
+        $metadata['description'] = wp_trim_words(htmlentities($single_event_information->description), Bizyhood_Core::META_DESCRIPTION_LENGTH, '');
+
+      }
+      
+      // promotions page metadata
+      if (is_page($promotions_page) && isset($wp_query->query_vars['bizyhood_id']) && $wp_query->query_vars['bizyhood_id'] != '') {
+        
+        
+        $bizyhood_name = '';
+        
+        $bizyhood_name = urldecode($wp_query->query_vars['bizyhood_name']);
+        $bizyhood_id = urldecode($wp_query->query_vars['bizyhood_id']);
+
+        
+        $single_promotion_information = Bizyhood_ApiCalls::single_business_additional_info('promotions', $bizyhood_name, $bizyhood_id);
+        
+        $business = '';
+        
+        if($single_promotion_information === false || !isset($single_promotion_information->name)) {
+          return $buffer;
+        }
+        
+        $metadata['title'] = htmlentities($single_promotion_information->name .', '. $single_promotion_information->business_name .' - '.get_bloginfo('name'));
+        $metadata['canonical'] = get_permalink($promotions_page) . $wp_query->query_vars['bizyhood_name'] .'/'.$wp_query->query_vars['bizyhood_id'].'/';
+        $metadata['description'] = wp_trim_words(htmlentities($single_promotion_information->details), Bizyhood_Core::META_DESCRIPTION_LENGTH, '');
+
+      }
+      
+      
+      if (empty($metadata)) {
```

Comment:
Beginning here, this could be the primary buffer replace method which takes the metadata object as the parameter. Then it can be called at the appropriate place for a given page. This way we avoid compounding the logic for every new page type, in this one giant method.